### PR TITLE
Refo 93 part2 - job approve/reject UI

### PR DESCRIPTION
**Pull request by kendebacker**

### CODE REVIEW

File: src/components/job/JobApproveReject.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+export default function JobApproveReject({ jobId }) {
+  const [hideFeedback, sethideFeedback] = useState(true);
+  const [jobFeedback, setJobFeedback] = useState("");
+  const { approveJob, rejectJob } = useJob();
+
+  return (
+    <div className="items-center text-center w-3/4">
+      {hideFeedback ? (
+        ""
```
Comment (makersmelx) [inline src/components/job/JobApproveReject.jsx:13]: `null` will be better

File: src/models/job.jsx [hunk 2]
```diff
@@ -56,8 +56,8 @@ const approveJob = async (jobId) => {
   await updateJob(jobId, { status: JOB_STATUS.APPROVED });
 };
 
-const rejectJob = async (jobId) => {
-  await updateJob(jobId, { status: JOB_STATUS.REJECTED });
+const rejectJob = async (jobId, adminMessage) => {
+  await updateJob(jobId, { status: JOB_STATUS.REJECTED, adminMessage: adminMessage });
```
Comment (makersmelx) [inline src/models/job.jsx:60]: just `{status: JOB_STATUS.REJECTED, adminMessage}` is ok

File: src/components/job/JobApproveReject.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        <div className="w-full">
+          <label className="label">
+            <span className="label-text">Reason for Rejection (Not Required)</span>
+          </label>
+          <textarea
+            onChange={(e) => setJobFeedback(e.target.value)}
+            className="textarea textarea-bordered w-full"
+          ></textarea>
+        </div>
+      )}
```
Comment (makersmelx) [inline src/components/job/JobApproveReject.jsx:24]: I would suggest use https://daisyui.com/components/modal/#

File: src/components/job/JobApproveReject.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          </button>
+        </div>
+      ) : (
+        <div className="mt-5">
+          <button onClick={() => sethideFeedback(true)} className="btn btn-warning">
+            Undo
+          </button>
+          <button
+            onClick={() => {
+              rejectJob(jobId, jobFeedback);
```
Comment (makersmelx) [inline src/components/job/JobApproveReject.jsx:46]: we may need a toast message when the action is done. So we could wrap them in the `useRequest`, see examples in job save(my new pr) or login

**Review by makersmelx — CHANGES_REQUESTED** [review]

File: src/components/job/JobApproveReject.jsx (hunk 3 above)
Comment (kendebacker) [reply src/components/job/JobApproveReject.jsx]: used modal!

File: src/components/job/JobApproveReject.jsx (hunk 4 above)
Comment (kendebacker) [reply src/components/job/JobApproveReject.jsx]: made toast messages for the approve/reject

**Review by makersmelx — CHANGES_REQUESTED** [review]
First is the primary button things
Second could be something like...
image
the Figma design seems to want the button to be at the bottom at the page.

what's more, this pic is for the refugee side, it has two buttons.

image

for admin, we have approve and reject, but we do not have the save button, so we are still having two buttons at the bottom of the page.

I guess, there should be a checking the user's role to decide to display a button or not

**Comment by kendebacker** [comment]
I made it so the "approved/reject" only show if its an admin. 

For the button colors, right now I have "Save", "Approve", and the modal "Reject" as primary and all the others as secondary, if thats incorrect could you let me know which should be which? Sorry, I don't think I'm understanding the color scheming completely. Thanks!

File: src/components/job/JobApproveReject.jsx [hunk 5]
```diff
@@ -40,7 +40,7 @@ export default function JobApproveReject({ jobId }) {
     <>
       <div className="items-center text-center w-3/4">
         <div className="w-full flex flex-row justify-center">
-          <label htmlFor="feedback-display" className="btn btn-primary">
+          <label htmlFor="feedback-display" className="btn btn-secondary">
```
Comment (makersmelx) [inline src/components/job/JobApproveReject.jsx:43]: sry for not making it clear. here should be a normal button, just `btn` is good

File: src/components/job/JobApproveReject.jsx [hunk 6]
```diff
@@ -67,10 +67,14 @@ export default function JobApproveReject({ jobId }) {
             ></textarea>
           </div>
           <div className="w-full flex flex-row justify-center mt-5">
-            <label htmlFor="feedback-display" className="btn btn-primary mr-5">
+            <label htmlFor="feedback-display" className="btn btn-secondary mr-5">
```
Comment (makersmelx) [inline src/components/job/JobApproveReject.jsx:70]: sry for not making it clear. here should be a normal button, just `btn` is good

Comment (makersmelx) [inline src/components/job/JobApproveReject.jsx:70]: just curious: what' the good for using label here? usually we are using `button`

**Review by makersmelx — APPROVED** [review]
Thanks! Just need to fix the color and everything is good to go~

File: src/components/job/JobApproveReject.jsx (hunk 5 above)
Comment (kendebacker) [reply src/components/job/JobApproveReject.jsx]: no worries, thanks for clarifying. done

File: src/components/job/JobApproveReject.jsx (hunk 6 above)
Comment (kendebacker) [reply src/components/job/JobApproveReject.jsx]: no worries, done

**Pull request merged**